src/uu/factor/src/factor.rs

```diff
@@ -96,7 +96,7 @@ impl fmt::Display for Factors {
         v.sort_unstable();
 
         let include_exponents = f.alternate();
-        for (p, exp) in v.iter() {
+        for (p, exp) in &*v {
```

Comment:
```suggestion
        for (p, exp) in v {
```

Reply:
yeah, just like you, i don't like the &* syntax :)
if v works, great! if it doesn't, i would prefer to keep v.iter

Reply:
The changes I suggested should work. But there are a few more cases of `&*` and `&mut *` left that I couldn't simplify. I'll go look for a clippy issue related to this. I'm guessing we aren't the only ones who don't like this.

Reply:
Found something related: https://github.com/rust-lang/rust-clippy/issues/11074

Reply:
Ok, I will simplify where possible and going back to `iter` (and disable the warning) in the other cases.

Reply:
Thanks!